src/mmrelay/setup_utils.py

```
@@ -546,10 +598,130 @@ def service_needs_update() -> tuple[bool, str]:
             "Service file uses legacy --config/--logfile flags (update to --home)",
         )
 
-    # Check if the service file is missing the --home flag entirely
-    # A valid v1.3 service file should have --home
-    if "--home" not in exec_start_line:
-        return True, "Service file is missing --home flag (v1.3 unified path model)"
+    # Accept either --home in ExecStart or MMRELAY_HOME in Environment for compatibility.
+    # Use token-level matching to avoid false positives on flags like --home-dir
+    has_home_flag = any(
+        token == "--home" or token.startswith("--home=") for token in exec_tokens
```

Ruff (0.15.7)

[error] Possible hardcoded password assigned to: "token"

(S105)